### PR TITLE
chore: app using api from env vars

### PR DESCRIPTION
**Pull request by fampinheiro**
To enable independent frontend and backend development the api url can be set using an environment variable. [Here](https://github.com/yldio/asap-hub/pull/81/files#diff-1eb6c386a33702c4811031f80affa3c0).

To do this in the CI a [script](https://github.com/yldio/asap-hub/pull/81/files#diff-2550986331811c521c3d9c78191f209f) is responsible for setting the `APP_HOSTNAME` and `API_HOSTNAME` explicitly.

### CODE REVIEW

File: ci/env-create.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
```
Comment (johnytiago) [inline ci/env-create.sh:1]: Are we sure this file is well-formed?

I was looking into the docs https://docs.gitlab.com/ee/ci/pipelines/job_artifacts.html#artifactsreportsdotenv.
> The .env file can’t have empty lines or comments (starting with #).

File: apps/react-templates/.gitlab-ci.yml [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
+    - setup
+    - build
   needs:
+    - setup
     - build
   variables:
-    <<: *tmpl_master_deploy_variables
-    ORIGIN: https://$BASE_URL
+    SLS_STAGE: production
+    APP_ORIGIN: https://$APP_ORIGIN
```
Comment (jeysal) [inline apps/react-templates/.gitlab-ci.yml:15]: This is very confusing, either it's with a protocol, then it's an `origin`, or it's without a protocol, then it's `host`. We shouldn't have `origin` sometimes include protocol and sometimes not. In this case, it seems like the reference should be `$APP_HOST`.

File: apps/frontend/src/config.ts [hunk 3]
```diff
@@ -1,20 +1,4 @@
...
-  case 4: {
-    baseUrl = `${window.location.protocol}//api-${window.location.host}`;
-    break;
-  }
-  default: {
-    baseUrl = 'http://localhost:3333/development';
-  }
-}
-export const API_BASE_URL = baseUrl.toString();
+export const API_BASE_URL =
```
Comment (jeysal) [inline apps/frontend/src/config.ts:3]: Should it not be `'http://localhost:3333/development'`?

File: apps/frontend/README.md [hunk 4]
```diff
@@ -12,6 +12,12 @@ Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
 The page will reload if you make edits.<br />
 You will also see any lint errors in the console.
 
+#### Environment Variables
+
+The frontend application uses the production API by default, to change this behavior you can set the `REACT_APP_API_BASE_URL` environment variable.
```
Comment (jeysal) [inline apps/frontend/README.md:17]: Does it? It seems like it defaults to `localhost`

File: apps/react-templates/.gitlab-ci.yml (hunk 2 above)
Comment (fampinheiro) [reply apps/react-templates/.gitlab-ci.yml]: The script exports just hostnames and they are used as is or with a protocol when used as origins.

File: apps/frontend/README.md (hunk 4 above)
Comment (fampinheiro) [reply apps/frontend/README.md:17]: Last minute change that wasn't properly updated. Thanks

**Pull request merged**